### DISCUSSION TITLE
Digital will / Deadman switch

### DISCUSSION THREAD

**Discussion by pawisoon** [dani-garcia/vaultwarden#3669] [Ideas]
Will it be possible to add functionality to share specific notes/passwords/all account contents, if we don't meet ping our Vault in a specific time period? 

Idea inspired by https://www.lastpass.com/features/digital-will

Cheers,
P

**Comment by BlackDex** [comment]
You probably are looking for https://bitwarden.com/help/emergency-access/

**Reply by alphafalcon** [reply]
I don't know how the referenced lastpass feature is implemented exactly, but I can imagine two distinct scenarios that meet different needs:

- Emergency Access: The recipient knows about the vault, already has an account and actively requests access
- Digital Will / Deadman's Switch: The recipient may be unaware the vault exists, does not have an account (that they know about), access is granted by some preconfigured condition.

Implementation-wise, both scenarios would probably use the same  emergency access mechanisms, no need to reinvent the wheel. What would be needed for the digital will feature would be "shadow" accounts that can be set up for recipients but are otherwise inactive until needed and some sort of trigger system.

One potential hazard is that the master passwords for the shadow accounts need to be stored somehow to get them to the recipients, as setting a new password is not possible without wiping the keys.
